android-json-form-wizard/src/main/java/com/vijay/jsonwizard/fragments/JsonWizardFormFragment.java

```diff
@@ -295,8 +326,8 @@ public void skipStepOnPreviousPressed() {
                 JSONObject formStep = getJsonApi().getmJSONObject().optJSONObject(JsonFormConstants.STEP + i);
                 if (formStep != null) {
                     checkIfStepIsBlank(formStep);
-                    if (shouldSkipStep()) {
-                        getFragmentManager().popBackStack();
+                    if (shouldSkipStep() && !stepHasNoSkipLogic(JsonFormConstants.STEP + i)) {
```

Comment:
why is this check added `!stepHasNoSkipLogic(JsonFormConstants.STEP + i)`

Reply:
The method checks if the previous step has fields with no skip logic, if it has then that step will not be skipped and vice versa